lib/msf/core/post/file.rb

```diff
@@ -34,9 +34,12 @@ def cd(path)
     e_path = expand_path(path) rescue path
     if session.type == "meterpreter"
       session.fs.dir.chdir(e_path)
+    elsif session.type == 'powershell'
+      cmd_exec("Set-Location -Path \"#{e_path}\"")
```

Comment:
What's the semantic difference between this and the existing `cd` functionality? 👀

Reply:
There is no difference actually, @smcintyre-r7 suggested me over slack to write powershell cmdlets separately so in the future we'll know it was tested and expected to work for powershell.

Reply:
so a should look like:
```
def method(params)
  if session.type == 'meterpreter'
    code_for_meterpreter
  elsif session.type == '"powershell"
    code_for_powershell
  else
     code_for_shell
   end
end
```

Reply:
I looked and `cd` is an alias for the `Set-Location` cmdlet in Powershell. You can verify that by running `Get-Help cd` and seeing the name is "Set-Location":

<details>
<summary>Get-Help output</summary>

```
PS C:\Windows\system32> Get-Help cd

NAME
    Set-Location
    
SYNOPSIS
    Sets the current working location to a specified location.
    
    
SYNTAX
    Set-Location [-LiteralPath] <string> [-PassThru] [-UseTransaction] [<Common
    Parameters>]
    
    Set-Location [[-Path] <string>] [-PassThru] [-UseTransaction] [<CommonParam
    eters>]
    
    Set-Location [-StackName <string>] [-PassThru] [-UseTransaction] [<CommonPa
    rameters>]
    
    
DESCRIPTION
    The Set-Location cmdlet sets the working location to a specified location. 
    That location could be a directory, a sub-directory, a registry location, o
    r another location stack.
    

RELATED LINKS
    Online version: http://go.microsoft.com/fwlink/?LinkID=113397
    about_Providers 
    Get-Location 
    Pop-Location 
    Push-Location 

REMARKS
    To see the examples, type: "get-help Set-Location -examples".
    For more information, type: "get-help Set-Location -detailed".
    For technical information, type: "get-help Set-Location -full".
```
</details>